src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -131,6 +153,72 @@ protected override void AddEmbeddedResourcesFromAddedModules(ArrayBuilder<Cci.Ma
         public override string Name => _metadataName;
         public AssemblyIdentity Identity => _sourceAssembly.Identity;
         public Version AssemblyVersionPattern => _sourceAssembly.AssemblyVersionPattern;
+
+        internal override SynthesizedAttributeData SynthesizeEmbeddedAttribute()
+        {
+            if ((object)_lazyEmbeddedAttribute != null)
+            {
+                return new SynthesizedAttributeData(
+                    _lazyEmbeddedAttribute.Constructor,
+                    ImmutableArray<TypedConstant>.Empty,
+                    ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
+            }
+
+            return base.SynthesizeEmbeddedAttribute();
+        }
+
+        protected override SynthesizedAttributeData TrySynthesizeIsReadOnlyAttribute()
+        {
+            if ((object)_lazyIsReadOnlyAttribute != null)
+            {
+                return new SynthesizedAttributeData(
+                    _lazyIsReadOnlyAttribute.Constructor,
+                    ImmutableArray<TypedConstant>.Empty,
+                    ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
+            }
+
+            return base.TrySynthesizeIsReadOnlyAttribute();
+        }
+
+        private void CreateEmbeddedAttributesIfNeeded()
+        {
+            if (_lazyEmbeddedAttributesDiagnostics.IsDefaultOrEmpty)
+            {
+                var diagnostics = DiagnosticBag.GetInstance();
+
+                if (this.NeedsGeneratedIsReadOnlyAttribute)
+                {
+                    CreateEmbeddedAttributeIfNeeded(
+                        ref _lazyEmbeddedAttribute,
+                        diagnostics,
+                        AttributeDescription.CodeAnalysisEmbeddedAttribute);
+
+                    CreateEmbeddedAttributeIfNeeded(
+                        ref _lazyIsReadOnlyAttribute,
+                        diagnostics,
+                        AttributeDescription.IsReadOnlyAttribute);
+                }
+
+                _lazyEmbeddedAttributesDiagnostics = diagnostics.ToReadOnlyAndFree();
+            }
+        }
+
+        private void CreateEmbeddedAttributeIfNeeded(ref SynthesizedEmbeddedAttributeSymbol symbol, DiagnosticBag diagnostics, AttributeDescription description)
+        {
+            if ((object)symbol == null)
```

Comment:
For now we are adding only `IsByRefLike`, so there is no need for `#2` just yet.
The current code seems ok to me for now.